src/test/backend/replicasetlist_test.go

```diff
@@ -15,10 +15,194 @@
+			"my-service 8080/TCP"},
+		{"my-service", "my-namespace", nil, "my-service.my-namespace"},
+		{"my-service", "my-namespace",
+			[]api.ServicePort{{Name: "foo", Port: 8080, Protocol: "TCP"}},
+			"my-service.my-namespace 8080/TCP"},
+	}
+	for _, c := range cases {
+		actual := getInternalEndpoint(c.serviceName, c.namespace, c.ports)
+		if actual != c.expected {
+			t.Errorf("getInternalEndpoint(%+v, %+v, %+v) == %+v, expected %+v",
+				c.serviceName, c.namespace, c.ports, actual, c.expected)
+		}
+	}
+}
+
+func TestGetMatchingServices(t *testing.T) {
+	cases := []struct {
+		services   []api.Service
+		replicaSet *api.ReplicationController
+		expected   []api.Service
+	}{
+		{nil, nil, nil},
+		{
+			[]api.Service{{Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name"}}}},
+			&api.ReplicationController{
+				Spec: api.ReplicationControllerSpec{Selector: map[string]string{"app": "my-name"}}},
+			[]api.Service{{Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name"}}}},
+		},
+		{
+			[]api.Service{
+				{Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name"}}},
+				{Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name", "ver": "2"}}},
+			},
+			&api.ReplicationController{
+				Spec: api.ReplicationControllerSpec{Selector: map[string]string{"app": "my-name"}}},
+			[]api.Service{{Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name"}}}},
+		},
+	}
+	for _, c := range cases {
+		actual := getMatchingServices(c.services, c.replicaSet)
+		if !reflect.DeepEqual(actual, c.expected) {
+			t.Errorf("getMatchingServices(%+v, %+v) == %+v, expected %+v",
+				c.services, c.replicaSet, actual, c.expected)
+		}
+	}
+}
+
+func TestGetReplicaSetList(t *testing.T) {
+	cases := []struct {
+		replicaSets []api.ReplicationController
+		services    []api.Service
+		expected    *ReplicaSetList
+	}{
+		{nil, nil, &ReplicaSetList{}},
+		{
+			[]api.ReplicationController{
+				{
+					Spec: api.ReplicationControllerSpec{
+						Selector: map[string]string{"app": "my-name-1"},
+						Template: &api.PodTemplateSpec{
+							Spec: api.PodSpec{Containers: []api.Container{{Image: "my-container-image-1"}}},
+						},
+					},
+				},
+				{
+					Spec: api.ReplicationControllerSpec{
+						Selector: map[string]string{"app": "my-name-2", "ver": "2"},
+						Template: &api.PodTemplateSpec{
+							Spec: api.PodSpec{Containers: []api.Container{{Image: "my-container-image-2"}}},
+						},
+					},
+				},
+			},
+			[]api.Service{
+				{
+					Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name-1"}},
+					ObjectMeta: api.ObjectMeta{
+						Name:      "my-app-1",
+						Namespace: "namespace-1",
+					},
+				},
+				{
+					Spec: api.ServiceSpec{Selector: map[string]string{"app": "my-name-2", "ver": "2"}},
+					ObjectMeta: api.ObjectMeta{
+						Name:      "my-app-2",
+						Namespace: "namespace-2",
+					},
+				},
+			},
+			&ReplicaSetList{
```

Comment:
Maybe we could delegate object creation to separate builder function? Service can be initialized by `createService("app", "my-name", "my-app", "namespace")` and then test will be easier to understand because this json-like format is sometimes hard to follow.